doc/sphinx/make-your-own-compositor.md

```diff
@@ -0,0 +1,248 @@
+---
+discourse: 4911,5164,5603,6756,8037
+---
+
+# Using Mir Libraries to Build Your Own Wayland Compositor
+If you've seen [the demo tutorial](explore-mir-using-demos), you're probably
+taken away by Mir's capabilities and want to use it and the supporting
+libraries to build your own Wayland compositor. This section will guide you
+through the installation and basic usage of Mir. You'll be even more impressed
+by how easy it is to create a Wayland compositor using Mir!
+
+### A barebones Mir compositor
+
+The following code block is the bare minimum you need to run a Mir compositor:
```

Comment:
We should also use some more "tutorial"-y language. For example, we could say "First, install Mir from the archive... Then, create a new file called `main.cpp` and write the following... Afterwards, create a new file called `CMakeLists.txt` and write the following... Finally, run these commands to build your project."

See https://diataxis.fr/tutorials/ for more details. In a nutshell, giving precise commands leaves no room for ambiguity

Reply:
I don't think the audience we're aiming for already knows the basics of C++. Making instructions more detailed might make things just noisier for readers

Reply:
Yup I agree with that! But the more direct the tutorial is, the better in my opinion. Tutorials are suited for imperative language and a lot of "hand-holding", even when it may be obvious to an experienced user.

I always find that the best tutorials allow me to copy-and-paste most of it to get a minimum viable demo working and then I dive deeper into the completed tutorial afterward.

Reply:
I modified the tutorials a bit, what do you think about their current state?